### DISCUSSION TITLE
How to set the profile for waypoints generated by the simple planner?

### DISCUSSION THREAD

**Discussion by marip8** [tesseract-robotics/tesseract_planning#273] [Q&A] [answered]
We should consider adding a string field to the simple planner profile for specifying the profile name to be assigned to the generated intermediate waypoints. Currently the profile of the generated waypoints is set from one of the parent waypoints (the target parent waypoint if I remember correctly).

One currently unsupported application for this change would be for a Descartes/Trajopt workflow where no sampling is allowed for the start and end waypoints, but axial sampling is allowed for any intermediate waypoints.

**Comment by Levi-Armstrong** [comment, marked as answer]
This is the reason the path profile was added. In the simple planner all interpolated state are assigned the path profile but the initial state remains the same. Does this not work for you use case?

**Reply by marip8** [reply]
I think that should work. I forgot that this was added to address this issue. It is still a bit unclear whose profile the generated waypoints inherit - the source or the target waypoint? It could be a little more intuitive to directly tell the simple planner what profile name to use for any newly generated waypoints

**Reply by Levi-Armstrong** [reply]
![image](https://user-images.githubusercontent.com/9803128/215636922-0d125ba1-24fc-4ad4-9671-8c5bd9402aa1.png)
